src/Way/Type.hs

```diff
@@ -0,0 +1,84 @@
+module Way.Type where
+
+import Data.IntSet (IntSet)
+import qualified Data.IntSet as Set
+import Data.List
+import Data.Maybe
+import Development.Shake.Classes
+import Hadrian.Utilities
```

Comment:
Again, it seems that `Hadrian.Utilities` is the only potential source of cyclic imports. But does it really cause a cyclic import? If not, perhaps we don't need this split?